Toolset/palettes/script editor/behaviors/revseeditorbehavior.livecodescript

```diff
@@ -5424,3 +5455,358 @@ private command EditorFieldCacheSwap pOldRuggedObjectId, pNewRuggedObjectId
+      unfoldAll
+   end if
+   
+   local tHandlers
+   local tScriptHTML
+   local tNumbersHTML
+   local tScriptNewHTML
+   local tNumbersNewHTML
+   local tHandlerStart
+   local tHandlerEnd
+   local tHandlerLastEnd
+   local tHidden
+   local tBeginU
+   local tEndU
+   local tScriptTemp
+   local tNumbersTemp
+   local tUnderlineText
+   
+   put " hidden" into tHidden
+   put "<u>" into tBeginU
+   put "</u>" into tEndU
+   put 0 into tHandlerLastEnd
+   
+   local tCompilationResult
+   actionCompile
+   put the result into tCompilationResult
+   if tCompilationResult <> empty then
+      then exit foldAll
+   end if
+   
+   put the cHandlers of field "Handlers" of group "Left Handler List" of group "Left Bar" into tHandlers
+   --    getHandlerList
+   --    put the result into tHandlers
+   put the htmlText of field "Script" of me into tScriptHtml
+   put the htmlText of field "Numbers" of me into tNumbersHtml
+   
+   repeat for each line aLine in tHandlers
+      put word 3 of aLine into tHandlerStart
+      put word 4 of aLine into tHandlerEnd
+      put line tHandlerLastEnd +1 to tHandlerStart - 1 of tScriptHTML & return after tScriptNewHTML
+      put line tHandlerLastEnd + 1 to tHandlerStart of tNumbersHTML & return after tNumbersNewHTML
+      put line tHandlerStart to tHandlerEnd of tScriptHTML into tScriptTemp
+      put line tHandlerStart + 1 to tHandlerEnd of tNumbersHTML into tNumbersTemp
+      put line 1 of tScriptTemp into tUnderLineText
+      
+      put char 1 to 3 of tUnderlineText & tBeginU & char 4 to - 5 of tUnderlineText & tEndU & char -4 to - 1 of tUnderlineText & cr  after tScriptNewHTML 
+      delete line 1 of tScriptTemp
+      repeat for each line aLine in tScriptTemp
+         put char 1 to 2 of aLine & tHidden & char 3 to - 1 of aLine & cr after tScriptNewHTML
+      end repeat
+      repeat for each line aLine in tNumbersTemp
+         put char 1 to 2 of aLIne &  tHidden & char 3 to - 1 of aLine & cr after tNumbersNewHTML
+      end repeat
+      put tHandlerEnd into tHandlerLastEnd
+   end repeat
+   put line tHandlerLastEnd + 1 to - 1 of tNumbersHtml after tNumbersNewHTML
+   put line tHandlerLastEnd + 1 to - 1 of tScriptHTML after tScriptNewHTML
+   set the htmlText of field "Script" of me to tScriptNewHTML
+   set the htmlText of field "Numbers" of me to tNumbersNewHTML
+end foldAll
+
+on syncFoldedFieldNumbers
+   local tHasHidden
+   put the hidden of line 1 to - 1 of field "Script" of me into tHasHidden
+   if tHasHidden is false then
+      exit syncFoldedFieldNumbers
+   end if
+   lock screen
+   local tScriptA
+   local tNumbersHTML
+   local tStart
+   local tEnd
+   local tStartHidden
+   local tKeys
+   
+   put "<p>" into tStart
+   put "</p>" into tEnd
+   put "<p hidden>" into tStartHidden
+   
+   put the styledText of field "Script" into tScriptA
+   put the keys of tScriptA into tKeys
+   sort tKeys ascending numeric
+   
+   repeat for each line aKey in tKeys
```

Comment:
I would have loved to do that but
Bug 18403 - [styledText] Hidden property of a line is not transferred with styledText
i.e. if you set the hidden to true in the styledText and reapply it it will not work.
And in my experience doing html on the fly is a tad faster than working on the styled text.